vm/rust/src/lib.rs

```diff
@@ -528,16 +530,16 @@ lazy_static! {
 #[allow(static_mut_refs)]
 fn get_versioned_constants(version: *const c_char) -> VersionedConstants {
     let version_str = unsafe { CStr::from_ptr(version) }.to_str().unwrap();
-    let version = StarknetVersion::from_str(&version_str)
-        .unwrap_or(StarknetVersion::from_str(&"0.0.0").unwrap());
+    let version = StarknetVersion::from_str(version_str)
+        .unwrap_or(StarknetVersion::from_str("0.0.0").unwrap());
```

Comment:
why do we have unwraps inside an unwrap_or

Can we have matches across the version strings that go straight to the StarknetVersion with a default branch for that returns the value for starknetversion value for "0.0.0"?

_currently a single missed typo can cause runtime errors_